src/hotspot/os/linux/os_linux.cpp

```diff
@@ -216,22 +216,27 @@ bool os::is_containerized() {
 }
 
 bool os::Container::memory_limit(physical_memory_size_type& value) {
-  assert(is_containerized(), "must be running containerized");
-  return OSContainer::memory_limit_in_bytes(value);
+  if (OSContainer::memory_limit_in_bytes(value) && value != value_unlimited) {
+    return true;
+  }
+  return false;
 }
 
 bool os::Container::memory_soft_limit(physical_memory_size_type& value) {
-  assert(is_containerized(), "must be running containerized");
-  return OSContainer::memory_soft_limit_in_bytes(value);
+  if (OSContainer::memory_soft_limit_in_bytes(value) && value != 0) {
+    return true;
+  }
+  return false;
 }
 
 bool os::Container::memory_throttle_limit(physical_memory_size_type& value) {
-  assert(is_containerized(), "must be running containerized");
-  return OSContainer::memory_throttle_limit_in_bytes(value);
+  if (OSContainer::memory_throttle_limit_in_bytes(value) && value != value_unlimited) {
+    return true;
+  }
+  return false;
```

Comment:
Same here with the changed `value` when `value_unlimited` is being read, and function returning `false`.